streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1016,6 +1015,9 @@ private void addRecordsToTasks(final ConsumerRecords<byte[], byte[]> records) {
                 throw new TaskMigratedException(task);
             }
 
+            if (task.streamTime() == -1) {
```

Comment:
Why `-1`？Could we define a constant referring to it?

Reply:
Why do we call `initializeTaskTime` in `addRecordsToTasks()` -- in a previous version it was called in `initializeNewTasks()` what seems to be more appropriate -- why did you move it?